test/Base_Tests/src/Data/Round_Spec.enso

```diff
@@ -24,7 +24,8 @@ type Batch_Runner
         epsilon_vec  = configs.map c-> c.at 3
         warning_vec  = configs.map c-> c.at 4
         got_vec = self.run_batch values_vec dps_vec rounding_mode
-        got_vec.each_with_index ix-> got->
+        0.up_to got_vec.length . each \ix ->
+            got = got_vec.at ix
```

Comment:
This fix is necessary to workaround [all the transitive closure of problems](https://github.com/enso-org/enso/pull/13118#issuecomment-2896884673).

Reply:
With 388d5736b4 the change 8ecb0a0 is no longer needed, which is great!

It makes me feel way more confident. There shall be no problems with warnings, when the _type check_ returns the _same value_ - e.g. with plain type checks. Only when doing conversions the warnings _get reattached_ - e.g. only then we could see some divergence in the behavior.